### PR TITLE
feat: Generalize ERC20Token class

### PR DESCRIPTION
**Pull request by tamaralipows**
- ERC20Token was used in instances that were not EVM-specific, which is bad organization.
- We decided that creating a token generic trait was overkill - especially since the main issue was just the Bytes. Instead, we/I decided that a good and simple solution was to:

1. Rename ERC20Token -> Token
2. Make the address Bytes instead of Address
3. Create a helper method to transform the Bytes to Address when needed

### CODE REVIEW

File: src/protocol/models.rs [hunk 1]
```diff
@@ -63,7 +63,7 @@ pub trait TryFromWithBlock<T> {
     async fn try_from_with_block(
         value: T,
         block: Header,
-        all_tokens: HashMap<Address, ERC20Token>,
+        all_tokens: HashMap<Address, Token>,
```
[the remark this answers is not in the document]
Comment (tamaralipows) [reply src/protocol/models.rs:66]: Good catch!!

File: src/models.rs [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 //!
 //! This module contains basic models that are shared across many
-//! components of the crate, including ERC20Token, Swap and SwapSequence.
+//! components of the crate, including Token, Swap and SwapSequence.
 //!
-//! ERC20Tokens provide instructions on how to handle prices and amounts,
+//! Tokens provide instructions on how to handle prices and amounts,
 //! while Swap and SwapSequence are usually used as results types.
-use alloy_primitives::{Address, U256};
+use alloy_primitives::{bytes::Bytes, U256};
```
Comment (zizou0x) [inline src/models.rs:8]: Shouldn't we use our own `tycho_core::Bytes`?

File: src/evm/protocol/erc20.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+pub fn bytes_to_erc20_address(address: &Bytes) -> Result<Address, SimulationError> {
+    if address.len() == 20 {
+        Ok(Address::from_slice(address))
+    } else {
+        Err(SimulationError::InvalidInput(
+            format!("Invalid ERC20 token address: {:?}", address),
+            None,
+        ))
+    }
+}
```
Comment (zizou0x) [inline src/evm/protocol/erc20.rs:17]: I think we should consider making a trait for this in `tycho-ethereum` (like we did for ethers type [here](https://github.com/propeller-heads/tycho-indexer/blob/82f9974a864a5d4ae7f216e64b74c5f73c020229/tycho-ethereum/src/lib.rs#L51-L127)) so we can reuse it, for example in the token-quoter repo

File: src/evm/protocol/erc20.rs
[the remark this answers is not in the document]
Comment (tamaralipows) [reply src/evm/protocol/erc20.rs:1]: nope because its also used for usv2 and 3

File: src/evm/protocol/erc20.rs (hunk 3 above)
Comment (tamaralipows) [reply src/evm/protocol/erc20.rs:18]: Nice idea - not urgent for the release though I guess?

**Review by dianacarvalho1 — APPROVED** [review]
Thank you @tamaralipows 🙏🏼 looks good to me!

File: examples/explorer/main.rs [hunk 4]
```diff
@@ -39,31 +39,21 @@ async fn main() {
...
-    // let (tick_tx, mut tick_rx) = mpsc::channel::<BlockState>(12);
-
-    // let consumer_handle = task::spawn(async move {
-    //     while let Some(message) = tick_rx.recv().await {
-    //         println!("got message: {:?}", message);
-    //     }
-    //     Ok(())
-    // });
-    // let tasks = [tycho_message_processor, consumer_handle];
+    let tasks = [tycho_message_processor, consumer_handle];
```
Comment (tamaralipows) [inline examples/explorer/main.rs:56]: Oops. Reverting this file. Ignore this.

**Review by zizou0x — APPROVED** [review]

**Pull request merged**